src/server/scripts/Northrend/FrozenHalls/PitOfSaron/pit_of_saron.cpp

```diff
@@ -246,10 +247,138 @@ class spell_trash_npc_glacial_strike : public SpellScriptLoader
+            {
+                me->SetReactState(REACT_PASSIVE);
+                me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
+            }
+
+            void IsSummonedBy(Unit* summoner) override
+            {
+                _summonerGUID = summoner->GetGUID();
+
+                _scheduler.Schedule(Seconds(3), [this](TaskContext /*context*/)
+                {
+                    DoCastSelf(SPELL_ICICLE_FALL_TRIGGER, true);
+                    DoCastSelf(SPELL_ICICLE_FALL_VISUAL);
+                    
+                    if (Unit* caster = ObjectAccessor::GetUnit(*me, _summonerGUID))
+                        caster->RemoveDynObject(SPELL_ICICLE_SUMMON);
+                });
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                _scheduler.Update(diff);
+            }
+
+        private:
+            TaskScheduler _scheduler;
+            ObjectGuid _summonerGUID;
+        };
+
+        CreatureAI* GetAI(Creature* creature) const override
+        {
+            return GetPitOfSaronAI<npc_pit_of_saron_icicleAI>(creature);
+        }
+};
+
+
+class spell_pos_ice_shards : public SpellScriptLoader
+{
+    public:
+        spell_pos_ice_shards() : SpellScriptLoader("spell_pos_ice_shards") { }
+
+        class spell_pos_ice_shards_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_pos_ice_shards_SpellScript);
+
+            bool Load() override
+            {
+                // This script should execute only in Pit of Saron
+                if (InstanceMap* instance = GetCaster()->GetMap()->ToInstanceMap())
+                    if (instance->GetInstanceScript())
+                        if (instance->GetScriptId() == sObjectMgr->GetScriptId(PoSScriptName))
+                            return true;
+
+                return false;
+            }
+
+            void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+            {
+                if (GetHitPlayer())
+                    GetCaster()->GetInstanceScript()->SetData(DATA_ICE_SHARDS_HIT, 1);
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_pos_ice_shards_SpellScript::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCHOOL_DAMAGE);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_pos_ice_shards_SpellScript();
+        }
+};
+
+class at_pit_cavern_entrance : public AreaTriggerScript
+{
+    public:
+        at_pit_cavern_entrance() : AreaTriggerScript("at_pit_cavern_entrance") { }
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+        {
+            if (InstanceScript* instance = player->GetInstanceScript())
+            {
+                if (Creature* tyrannus = ObjectAccessor::GetCreature(*player, instance->GetGuidData(DATA_TYRANNUS)))
+                {
+                    //tyrannus->AI()->Talk(1);
```

Comment:
Travis complain about tyrannus not used.
btw, this code should not be commented (will not crash if text is not present yet)